src/tree-and-graph/paths-with-sum.ts

```diff
@@ -0,0 +1,48 @@
+// page 123(111) Q-4.12
+
+import { BTNode } from "../types";
+
+function getPathsWithSum(
+  root: BTNode<number> | undefined,
+  sum: number,
+  path: BTNode<number>[],
+  pathsWithSum: BTNode<number>[][]
+) {
+  if (!root) {
+    return;
+  }
+  path.push(root);
+  const remain = sum - root.val;
+  if (remain === 0) {
+    pathsWithSum.push([...path]);
+  }
+  getPathsWithSum(root.left, remain, path, pathsWithSum);
+  getPathsWithSum(root.right, remain, path, pathsWithSum);
+  path.pop();
+
+  return;
+}
+
+export function main(root: BTNode<number>, sum: number): number {
```

Comment:
better name suggestion: `findPathsWithSum`. 

Exported function/class/object name should match with the file name.